File: projects/lostengine/lost/lua/lost/io/rg/DefaultFrameBuffer.lua
module("lost.io", package.seeall)

require("lost.io.Loader")

-- calls the glClear command with the given mask
function Loader:DefaultFrameBuffer(definition)
  local result = lost.rg.DefaultFrameBuffer.create()
  
  local active = definition["active"]
  local name = definition["name"]
  if active ~= nil then
    result.active = active
  end
  if name ~= nil then
    result.name = name
  end
  return result
end

File: projects/lostengine/lost/rg/DefaultFrameBuffer.cpp
#include "lost/rg/DefaultFrameBuffer.h"
#include "lost/gl/Context.h"

namespace lost
{
namespace rg
{

DefaultFrameBufferPtr DefaultFrameBuffer::create()
{
  return DefaultFrameBufferPtr(new DefaultFrameBuffer);
}

void DefaultFrameBuffer::process(gl::ContextPtr& ctx)
{
  if(!active) return;
  ctx->bindDefaultFramebuffer();
}

}
}